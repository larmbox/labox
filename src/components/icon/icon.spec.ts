import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { LIcon } from '~/index';

test('mount component', async () => {
  expect(LIcon).toBeTruthy();

  const component = mount(LIcon, {
    props: {
      id: 'id',
      icon: 'some-icon',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain('icon');
});

test('i alias', async () => {
  const component = mount(LIcon, {
    props: {
      id: 'id',
      i: 'some-icon',
    },
  });

  expect(component.html()).toMatchSnapshot();
});

test('types', () => {
  const component1 = mount(LIcon, {
    props: {
      prefix: 'i-',
      type: 'inline',
      icon: 'some-icon',
    },
  });

  const component2 = mount(LIcon, {
    props: {
      prefix: 'i-',
      type: 'class',
      icon: 'some-icon',
    },
  });

  expect(component1.text()).toEqual('i-some-icon');
  expect(component2.classes()).toEqual(['icon', 'i-some-icon']);
});

test('inherit attributes', async () => {
  const component = mount(LIcon, {
    props: { notaprop: '1', prefix: 'i-' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('prefix="i-"');
});
