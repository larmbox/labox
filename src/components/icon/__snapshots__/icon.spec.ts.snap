// Vitest Snapshot v1

exports[`i alias 1`] = `"<span class=\\"icon some-icon\\" aria-hidden=\\"true\\"></span>"`;

exports[`mount component 1`] = `"<span class=\\"icon some-icon\\" aria-hidden=\\"true\\"></span>"`;
